src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -136,24 +141,13 @@ private enum Flags : byte
                     Debug.Assert(memberName == DefaultIndexerName);
                 }
 
-                _name = ExplicitInterfaceHelpers.GetMemberName(WellKnownMemberNames.Indexer, _explicitInterfaceType, aliasQualifierOpt);
+                _name = ExplicitInterfaceHelpers.GetMemberName(explicitInterfaceMemberInfo?.ExplicitInterfaceSpecifier, WellKnownMemberNames.Indexer);
             }
             else
             {
                 _name = _lazySourceName = memberName;
             }
 
-            if ((isAutoProperty && hasGetAccessor) || hasInitializer)
```

Comment:
>if ((isAutoProperty && hasGetAccessor) || hasInitializer) [](http://example.com/codeflow?start=12&length=57)

It is not clear why this operation is delayed. It also doesn't look like this is going to make much difference. The backing field is also in the list of members for a type and it is going to be requested right after the property is created.